lib/chef_zero/server.rb

```diff
@@ -238,67 +240,78 @@ def start(publish = true)
     # to the forked process.
     #
     # @param [Fixnum] wait
-    #   the number of seconds to wait for the server to start
+    #   the number of seconds to wait for the server to start (nil or 0 means forever)
     #
     # @return [Thread]
     #   the thread the background process is running in
     #
-    def start_background(wait = 5)
-      @server = WEBrick::HTTPServer.new(
-        :DoNotListen => true,
-        :AccessLog   => [],
-        :Logger      => WEBrick::Log.new(StringIO.new, 7),
-        :SSLEnable  => options[:ssl],
-        :SSLCertName  => [ [ 'CN', WEBrick::Utils::getservername ] ],
-        :StartCallback => proc {
-          @running = true
-        }
-      )
+    # @raise [ServerAlreadyStartedError] if the server is already started.
+    #
+    def start_background(wait = nil)
+      server_mutex.synchronize do
+        if !server.nil?
+          raise ServerAlreadyStartedError
+        end
+        self.server = WEBrick::HTTPServer.new(
+          :DoNotListen => true,
+          :AccessLog   => [],
+          :Logger      => WEBrick::Log.new(StringIO.new, 7),
+          :SSLEnable  => options[:ssl],
+          :SSLCertName  => [ [ 'CN', WEBrick::Utils::getservername ] ],
+          :StartCallback => proc {
+            self.running = true
+          }
+        )
+      end
+
       ENV['HTTPS'] = 'on' if options[:ssl]
-      @server.mount('/', Rack::Handler::WEBrick, app)
+      server.mount('/', Rack::Handler::WEBrick, app)
```

Comment:
Right - because we have the mutex and the server check, I would think only 1 server would exist at this point